newIDE/app/src/InstancesEditor/FullSizeInstancesEditorWithScrollbars.js

```diff
@@ -33,154 +79,394 @@ type Props = {|
+        ) {
+          hideScrollbarsAfterDelay();
+        }
+        document.removeEventListener('mousemove', mouseMoveHandler);
+        document.removeEventListener('mouseup', mouseUpHandler);
       },
-      () => {
-        if (this._editor) {
-          this._editor.scrollTo(this.state.xValue, this.state.yValue);
+    [hideScrollbarsAfterDelay]
+  );
+  const makeMouseUpYThumbHandler = React.useCallback(
+    mouseMoveHandler =>
+      function mouseUpHandler(e: MouseEvent) {
+        isDragging.current = false;
+        // If the user releases the mouse outside of the detection zone, we want to hide the scrollbars.
+        if (
+          e.target !== yScrollbarDetectionZone.current &&
+          e.target !== yScrollbarTrack.current &&
+          e.target !== yScrollbarThumb.current
+        ) {
+          hideScrollbarsAfterDelay();
         }
+        document.removeEventListener('mousemove', mouseMoveHandler);
+        document.removeEventListener('mouseup', mouseUpHandler);
+      },
+    [hideScrollbarsAfterDelay]
+  );
+
+  // When the user clicks on the thumbs, we register new events:
+  // - mousemove: to move the thumb
+  // - mouseup: to stop moving the thumb
+  // We also note down where the user clicked on the thumb, so that we can
+  // move the thumb relative to this position.
+  const mouseDownXThumbHandler = React.useCallback(
+    (e: MouseEvent) => {
+      isDragging.current = true;
+      const initialDragPosition = e.clientX;
+      // xStartScrollValue.current = xValue.current;
+      const mouseMoveHandler = makeMouseMoveXHandler(
+        initialDragPosition,
+        xValue.current
+      );
+      const mouseUpHandler = makeMouseUpXThumbHandler(mouseMoveHandler);
+
+      document.addEventListener('mousemove', mouseMoveHandler);
+      document.addEventListener('mouseup', mouseUpHandler);
+    },
+    [makeMouseMoveXHandler, makeMouseUpXThumbHandler]
+  );
+  const mouseDownYThumbHandler = React.useCallback(
+    (e: MouseEvent) => {
+      isDragging.current = true;
+      const initialDragPosition = e.clientY;
+      const mouseMoveHandler = makeMouseMoveYHandler(
+        initialDragPosition,
+        yValue.current
+      );
+      const mouseUpHandler = makeMouseUpYThumbHandler(mouseMoveHandler);
+
+      document.addEventListener('mousemove', mouseMoveHandler);
+      document.addEventListener('mouseup', mouseUpHandler);
+    },
+    [makeMouseMoveYHandler, makeMouseUpYThumbHandler]
+  );
+
+  // Add the mouse down events once on mount.
+  React.useEffect(
+    () => {
+      const xScrollbarThumbElement = xScrollbarThumb.current;
+      const yScrollbarThumbElement = yScrollbarThumb.current;
+      if (!xScrollbarThumbElement || !yScrollbarThumbElement) return;
+
+      xScrollbarThumbElement.addEventListener(
+        'mousedown',
+        mouseDownXThumbHandler
+      );
+      yScrollbarThumbElement.addEventListener(
+        'mousedown',
+        mouseDownYThumbHandler
+      );
+    },
```

Comment:
Should you unregister the event listener? Or don't you do it because it will be destroyed when the element is destroyed?

Reply:
Tested. It is removed from the DOM only when the editor tab is closed, and the event listeners are removed because destroyed. 👍

Reply:
tested when changing from one editor to the other, no conflict.